modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
What if we were to require the **primary type** have `mode:unorderedCollection` as a base? That might even simplify the code a bit, since we'd never have to check mixins.

Anyway, isn't it possible to change the primary type of an item?